src/dp_flow.c

```diff
@@ -100,6 +100,7 @@ static __rte_always_inline int dp_build_icmp_flow_key(const struct dp_flow *df,
 			return DP_ERROR;
 		}
 
+		// changing l3_dst and l3_src without setting l3_type to IPV4
 		key->l3_dst.ip4 = ntohl(icmp_err_ip_info.err_ipv4_hdr->src_addr);
 		key->l3_src.ip4 = ntohl(icmp_err_ip_info.err_ipv4_hdr->dst_addr);
```

Comment:
Now this one can be "solved" by a comment if this code only fires for IPv4

Reply:
Yes, fired only in ICMP path which is ipv4 only.